src/KafkaKeyValue.ts

```diff
@@ -207,53 +231,53 @@ export default class KafkaKeyValue {
     this.fetchImpl = getFetchImpl(config);
     this.logger = getLogger({ name: `kkv:${this.getCacheName()}` });
 
-    updateEvents.on('update', async (requestBody: { v: number, offsets: { [partition: string]: number }, topic: string, updates: { [key: string]: {} } }) => {
-      if (requestBody.v !== 1) throw new Error(`Unknown kkv onupdate protocol ${requestBody.v}!`);
+    updateEvents.on('update', this.updateListener.bind(this));
+  }
 
-      const {
-        topic, offsets, updates
-      } = requestBody;
+  async updateListener(requestBody: UpdateRequestBody) {
+    if (requestBody.v !== 1) throw new Error(`Unknown kkv onupdate protocol ${requestBody.v}!`);
 
-      const expectedTopic = this.topic;
-      this.logger.trace({ topic, expectedTopic }, 'Matching update event against expected topic');
-      if (topic !== expectedTopic) {
-        this.logger.trace({ topic, expectedTopic }, 'Update event ignored due to topic mismatch. Business as usual.');
-        return;
-      }
+    const {
```

Comment:
Would it be possible to crash a kkv client by sending an onupdate with malformed or missing metadata?

Reply:
Yes, good catch. We don't introduce that in this PR though.

Reply:
The destructuring (below) is new though. But maybe tthat's not what poses the risk.

Reply:
I don't think it is though, the refactor I had to do just messed up the diff in this :-/